### PR TITLE
Addition of missing exported function prototype to osdp header files.

### PR DESCRIPTION
**Pull request by dmercer-google**
This adds osdp_set_command_complete_callback to osdp.h and  set_command_complete_callback to class Common in osdp.hpp.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,6 +103,7 @@ that are exposed by libosdp.
 git clone https://github.com/goToMain/libosdp --recurse-submodules
 # git submodule update --init (if you missed doing --recurse-submodules earlier)
 cd libosdp
+./configure.sh
```
Comment (sidcha) [inline README.md:106]: This project has two build systems: cmake and make. The `configure.sh` is used for the make build while this section of the document is describing the cmake build (and is right as it is). If you wish to document the make build, please do that in a separate section.

File: doc/libosdp/build-and-install.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ these with your application as needed (-losdp or -losdpstatic). Have a look at
 ``sample/*`` for details on how to consume this library.
 
 .. code:: sh
-
+    ./configure.sh
```
Comment (sidcha) [inline doc/libosdp/build-and-install.rst:10]: Same comment here too.

File: include/osdp.h [hunk 3]
```diff
@@ -943,6 +932,29 @@ int osdp_cp_modify_flag(osdp_t *ctx, int pd_idx, uint32_t flags, bool do_set);
...
+ * This callback is invoked when a reply to a command is received.
+ *
+ * @param ctx OSDP context
+ * @param cb The callback function's pointer
+ * @param arg A pointer that will be passed as the first argument of `cb`
+ */
+void osdp_set_command_complete_callback(osdp_t *ctx,
+					osdp_command_complete_callback_t cb,
+					void *arg);
+
```
Comment (sidcha) [inline include/osdp.h:957]: This API was marked deprecated and then removed [here](https://github.com/goToMain/libosdp/commit/0bde480c89fde2b317f2fad712333a3211b9b7eb#diff-8c3f6bb64d22c45f45f9f9735b80c1a6a104e7e9b369dd4a121cf8a03e0ffa22L1055-L1070) in preparation for a major release (v3.0.0) from master. But I see the removal was done hastily and left most of the implementation :(. I will remove the reset of this API in a bit.

Curious, what is your use of this API? It exposes low level OSDP commands to the user (which they should not have to care about). If you have a valid use-case, we can always expose a different API to cover that case.

Comment (Aalawfi) [reply include/osdp.h:957]: Just came across this yesterday, the use case in our application  currently need to kickoff some application specific tasks when CP receives a response for some OSDP command e.g. REPLY_PDID, REPLY_PDCAP..

**Comment by sidcha** [comment]
The doc issue is hopefully addressed in [f21c7d4](https://github.com/goToMain/libosdp/commit/f21c7d42abc00bf430a56f8a431ac2cae8634f0c). The use of command_complete_callback API described here is not good enough to resuscitate it.

Closing this PR.

File: include/osdp.h (hunk 3 above)
Comment (dmercer-google) [reply include/osdp.h:957]: No direct use case at the moment.  I just noticed the inconsistency when I was tracing through the callback flows.

**Pull request closed without merging**